code/modules/mob/mob.dm

```diff
@@ -194,7 +194,7 @@
 	var/hear_dist = 7
 	if(max_distance) hear_dist = max_distance
 	for(var/mob/M as anything in hearers(hear_dist, src.loc))
-		M.show_message(message, 2, message_flags = message_flags)
+		M.show_message(message, (isdeaf(M) ? 1 : 2), message_flags = message_flags)
```

Comment:
use the defines for showmessage

Reply:
also this isn't really the right approach, if you're deaf you should miss all audible messages. don't we want to change what's calling it